sharedbuffers/mapped_struct.py

```diff
@@ -1855,41 +1861,58 @@ def __contains__(self, elem):
 
         h2 = _stable_hash(elem)
         while lo < hi:
-            step = (lo + hi) >> 1
-            val = self.objlist._c_getitem(step, dcode, objlen, itemsize, offset, _struct, None)
+            mid = lo + ((hi - lo) >> 1)
+            val = self.objlist._c_getitem(mid, dcode, objlen, itemsize, offset, _struct, None)
             h1 = _stable_hash(val)
-            if h1 == h2 and val == elem:
-                return True
+            if h1 == h2:
+                # Equal hashes - Look in both directions for a match, without binary search
+                if val == elem:
+                    return True
+                while mid > 0:
+                    val = self.objlist._c_getitem(mid, dcode, objlen, itemsize, offset, _struct, None)
+                    h1 = _stable_hash(val)
+                    if h1 != h2:
+                        break
+                    elif val == elem:
+                        return True
+                    mid -= 1
+
+                mid = lo + ((hi - lo) >> 1)
+                hi = len(self.objlist)
```

Comment:
You don't need this. You already know there's no match behond `hi` whatever value it has.

Reply:
Hm. I was thinking of the most pathological case possible, in which all elements of the list have the same hash value, but are all different. In that case, I would have to look both to the left and the right ... right ?